### PR TITLE
Update lumi.plug to report total energy

### PR DESCRIPTION
**Pull request by Brazen00**
I've updated the xiaomi quirk for lumi.plug and added a metering cluster using the data from the second analog cluster. It works with my plug in my Docker install of HA.

The only issue that I'm still having is that the second analog sensor and the current_summation_delivered sensor in HA only update when I manually request the present value from the analog sensor via "Manage Clusters". This is different from the first analog sensor (power) and the power sensor from ElectricalMeasurementCluster which updates whenever the power drawn by the plug changes.

Is there a way of periodically polling the analog sensor within HA?

### CODE REVIEW

**Comment by Brazen00** [comment]
Fixes https://github.com/zigpy/zha-device-handlers/issues/1135

File: zhaquirks/xiaomi/__init__.py [hunk 1]
```diff
@@ -493,6 +494,24 @@ def _update_attribute(self, attrid, value):
...
+    cluster_id = AnalogInput.cluster_id
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        self._current_state = {}
+        super().__init__(*args, **kwargs)
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if value is not None and value >= 0:
```
Comment (dmulcahey) [inline zhaquirks/xiaomi/__init__.py:512]: Wouldn’t 0 be a valid value for this?

**Comment by dmulcahey** [comment]
I have the US plugs. I’ll try this later today and I’ll attempt to see what the deal is with them not updating

File: zhaquirks/xiaomi/__init__.py (hunk 1 above)
Comment (Brazen00) [reply zhaquirks/xiaomi/__init__.py:512]: Yes, that's covered by `value >= 0`

**Comment by Brazen00** [comment]
That would be great, thanks.

File: zhaquirks/xiaomi/__init__.py (hunk 1 above)
Comment (dmulcahey) [reply zhaquirks/xiaomi/__init__.py:512]: Sorry not enough coffee

**Comment by dmulcahey** [comment]
This needs to be looked at a bit more so I am going to leave this open for now.

**Comment by lvcabral** [comment]
I have the same issue, how do I test this patch on my Home Assistant instance?

**Comment by Brazen00** [comment]
Check out this info: https://github.com/zigpy/zha-device-handlers#testing-new-releases

**Comment by jpgimenez** [comment]
Hi, I own a xiaomi aqara plug, and tried this branch... it added 2 new entities, but no data is reported.

**Comment by Brazen00** [comment]
Can you try manually updating the present_value of AnalogInputCluster2 via "Manage Clusters"? That should update the value of the sensors. Getting this done automatically is the problem that still needs solving.

**Comment by jpgimenez** [comment]
> Can you try manually updating the present_value of AnalogInputCluster2 via "Manage Clusters"? That should update the value of the sensors. Getting this done automatically is the problem that still needs solving.

yes, you are right, it updates the value, only on the popup, doesn't update the sensor

**Comment by lvcabral** [comment]
Any idea when this can be merged and released?

**Comment by lvcabral** [comment]
One month later, any chance this fix to be added to a release?

**Comment by Brazen00** [comment]
I'd love for it to, but I've reached the limits of my knowledge and understanding. I think I know what would fix it (see https://github.com/zigpy/zha-device-handlers/issues/1251) but I don't know how to implement it.

**Comment by Brazen00** [comment]
Actually, I've just noticed https://github.com/zigpy/zha-device-handlers/pull/1226 which may help solve the problem. I'll have a play around with it when I get a chance.

**Comment by Brazen00** [comment]
OK so using the PolledCluster class from https://github.com/zigpy/zha-device-handlers/pull/1226 I've managed to get this solved. Just waiting for that to be implemented.

**Comment by Draluak** [comment]
fyi: while waiting for a permanent solution, the workaround mentioned in https://github.com/zigpy/zha-device-handlers/pull/1226 using [zha-toolkit](https://github.com/mdeweerd/zha-toolkit) combined with a time triggered automation works for my lumi.plug.mmeu01 devices. Custom 'event_success' is optional.

```
- alias: update xiaomi plug total consumption 
  trigger:
    - platform: time_pattern
      minutes: /30
  action:
    - service: zha_toolkit.attr_read
      data:
        ieee: sensor.miplug_fridge_consumption_total
        event_success: miplugupdated
        cluster: 0x000c
        attribute: 0x0055
        endpoint: 22
    - service: zha_toolkit.attr_read
      data:
        ieee: sensor.miplug_macbook_consumption_total
        event_success: miplugupdated
        cluster: 0x000c
        attribute: 0x0055
        endpoint: 22
    - service: zha_toolkit.attr_read
      data:
        ieee: sensor.miplug_playstation_consumption_total
        event_success: miplugupdated
        cluster: 0x000c
        attribute: 0x0055
        endpoint: 22
    - service: zha_toolkit.attr_read
      data:
        ieee: sensor.miplug_workstation_consumption_total
        event_success: miplugupdated
        cluster: 0x000c
        attribute: 0x0055
        endpoint: 22

**Comment by TheJulianJES** [comment]
Just to confirm, both US (``lumi.plug``) and EU variants (``lumi.plug.mmeu01``) need to have their ``current_summ_delivered`` attribute polled in your case?
(working on something similar with ``lumi.plug.maeu01``)

**Comment by TheJulianJES** [comment]
There was an alternative implementation in https://github.com/zigpy/zha-device-handlers/pull/2116 that should properly fix this.

That should make it into HA 2023.2.0 and later.

**Pull request closed without merging**